### PR TITLE
genericapiserver: unify swagger and openapi in config

### PR DESCRIPTION
**Pull request by sttts**
- make swagger config customizable
- remove superfluous `Config.Enable*` flags for OpenAPI and Swagger.

This is necessary for downstream projects to tweak the swagger spec.

### CODE REVIEW

**Comment by sttts** [comment]
@deads2k green and read for review

File: pkg/genericapiserver/config.go [hunk 1]
```diff
@@ -136,8 +135,11 @@ type Config struct {
 	// Serializer is required and provides the interface for serializing and converting objects to and from the wire
 	// The default (api.Codecs) usually works fine.
 	Serializer runtime.NegotiatedSerializer
-	// OpenAPIConfig will be used in generating OpenAPI spec.  This has "working" defaults.
-	OpenAPIConfig *common.Config
+	// OpenAPIConfig will be used in generating OpenAPI spec. This is nil by default. Use DefaultOpenAPICOnfig for "working" defaults.
+	OpenAPIConfig *openapicommon.Config
+	// SwaggerConfig will be used in generating Swagger spec. This has "working" defaults. Set it to nil to disable Swagger.
```
Comment (deads2k) [inline pkg/genericapiserver/config.go]: Why can one default on and the other has to default off?

**Comment by sttts** [comment]
@deads2k are you good with this now (modulo green unit tests) ?

File: pkg/genericapiserver/config.go [hunk 2]
```diff
@@ -136,8 +135,11 @@ type Config struct {
 	// Serializer is required and provides the interface for serializing and converting objects to and from the wire
 	// The default (api.Codecs) usually works fine.
 	Serializer runtime.NegotiatedSerializer
-	// OpenAPIConfig will be used in generating OpenAPI spec.  This has "working" defaults.
-	OpenAPIConfig *common.Config
+	// OpenAPIConfig will be used in generating OpenAPI spec. This is nil by default. Use DefaultOpenAPIConfig for "working" defaults.
+	OpenAPIConfig *openapicommon.Config
+	// SwaggerConfig will be used in generating Swagger spec. This is nil by default. Use DefaultSwaggerConfig fir "working" defaults.
```
Comment (deads2k) [inline pkg/genericapiserver/config.go]: s/fir/for/

**Comment by deads2k** [comment]
> @deads2k are you good with this now (modulo green unit tests) ?

you've got a typo, but yeah lgtm.

File: pkg/genericapiserver/config.go (hunk 2 above)
Comment (sttts) [inline pkg/genericapiserver/config.go]: fixed

**Pull request merged**